### PR TITLE
Added Network class for all the API calling methods

### PR DESCRIPTION
**Pull request by imjog**
I have added Network.swift file to the Project. I explained about this in #23 
This can be used for all the network calls. With this, we do not need to write Alamofire methods again and again.
I have added an example named `oauthExample` in `AuthService.swift` file.  Which can be removed after reviewing PR or can be used for implementation.
This was a simple example. With Network class we can easily handle complicated methods.
Also, I can explain how to call network methods in readme.md (or a new .md file).
@codedentwickler @cpg please review the PR! Thanks :)

### CODE REVIEW

**Comment by cpg** [comment]
Is this all necessary? Isn't there some http client library in wide use already?

Does it matter if it's in swift or obj-c?

Just wondering

File: AmahiAnywhere/AmahiAnywhere/Services/Network.swift [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+import Alamofire
+
+fileprivate struct Defaults {
+  static let timeoutInternval: TimeInterval = 20
+}
+
+class Network {
+
+  static let shared = Network()
+
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/Services/Network.swift]: Network class looks good. I think it is an overkill with what we have on the app at the moment. We can always add other implementations when we are using them as the put and patch. 
The .get and .post method are useful at the moment. I think whoever is implementing other methods can add them to suit the way the API implements these methods. 

Also, the network indicator should be managed in the presenter class making the network call. The UIViewController has already been extended so you can easily show and dismiss the loading dialog.  You should check out the base package and check what is there. There is also an API endpoint class for managing ApiEndpoints easily on the app.

File: AmahiAnywhere/AmahiAnywhere/Data/Remote/ApiCalls/AuthService.swift [hunk 2]
```diff
@@ -38,5 +38,23 @@ class AuthService {
...
+            if error != nil {
+                debugPrint(error.debugDescription)
+                completion(nil)
+                return
+            }
+            guard let data = parsedJSON else {
+                completion(nil)
+                return
+            }
+            let userAccessToken = data["access_token"].string
```
Comment (codedentwickler) [inline AmahiAnywhere/AmahiAnywhere/Data/Remote/ApiCalls/AuthService.swift]: I won't advice plucking data from JSON like this. We can create a model class for the response and EVReflection handles deserialization so it is easy to manage if we need to make changes.

File: AmahiAnywhere/AmahiAnywhere/Services/Network.swift (hunk 1 above)
Comment (imjog) [reply AmahiAnywhere/AmahiAnywhere/Services/Network.swift]: Hi @codedentwickler Thank you so much for reviewing :)
I added this class keeping common API calling methods in mind.  I am also agree with your points too.

**Comment by imjog** [comment]
@snuff4  I totally agree with your opinion. I made this PR now because we can go with this method or different method so that in future when enough implementation is done we don't have to change all the old implementation.

**Comment by imjog** [comment]
Currently used way is also good. I am just proposing a new way. For now, I can close the PR and reopen again if we need this in future. @codedentwickler @cpg @snuff4

**Pull request closed without merging**